singlefactorauth/src/main/java/com/web3auth/singlefactorauth/SingleFactorAuth.kt

```diff
@@ -1,191 +1,128 @@
-import org.torusresearch.torusutils.types.TypeOfUser
-import org.torusresearch.torusutils.types.VerifierArgs
+import org.torusresearch.torusutils.types.VerifierParams
+import org.torusresearch.torusutils.types.VerifyParams
+import org.torusresearch.torusutils.types.common.TorusKey
+import org.torusresearch.torusutils.types.common.TorusOptions
+import org.torusresearch.torusutils.types.common.TorusPublicKey
 import org.web3j.crypto.Hash
-import java.util.concurrent.CompletableFuture
-import java.util.concurrent.ExecutionException
 
-class SingleFactorAuth(singleFactorAuthArgs: SingleFactorAuthArgs) {
-    private var nodeDetailManager: FetchNodeDetails
+class SingleFactorAuth(sfaParams: SFAParams, ctx: Context) {
+    private var nodeDetailManager: FetchNodeDetails =
+        FetchNodeDetails(sfaParams.getNetwork())
     private val torusUtils: TorusUtils
-    private lateinit var sessionManager: SessionManager
+    private var sessionManager: SessionManager
     private val gson = GsonBuilder().disableHtmlEscaping().create()
+    private var network: Web3AuthNetwork
 
     init {
-        nodeDetailManager = FetchNodeDetails(singleFactorAuthArgs.getNetwork())
-        val opts = TorusCtorOptions("single-factor-auth-android", singleFactorAuthArgs.clientid)
-        opts.isEnableOneKey = true
-        opts.network = singleFactorAuthArgs.getNetwork().toString()
-        opts.signerHost =
-            SingleFactorAuthArgs.SIGNER_MAP[singleFactorAuthArgs.getNetwork()] + "/api/sign"
-        opts.allowHost =
-            SingleFactorAuthArgs.SIGNER_MAP[singleFactorAuthArgs.getNetwork()] + "/api/allow"
-        torusUtils = TorusUtils(opts)
+        val torusOptions = sfaParams.getClientId()?.let {
+            TorusOptions(
+                it, sfaParams.getNetwork(),
+                sfaParams.getNetworkUrl(), sfaParams.getServerTimeOffset(), true
+            )
+        }
+        network = sfaParams.getNetwork()
+        torusUtils = TorusUtils(torusOptions)
+        sessionManager = SessionManager(ctx)
     }
 
-    @RequiresApi(Build.VERSION_CODES.O)
-    @Throws(ExecutionException::class, InterruptedException::class)
-    fun getKey(
-        loginParams: LoginParams,
-        context: Context? = null,
-        sessionTime: Long = 86400
-    ): CompletableFuture<TorusKey> {
-        val torusKeyCompletableFuture: CompletableFuture<TorusKey> =
-            CompletableFuture<TorusKey>()
+    fun initialize(ctx: Context): SFAKey {
+        val data = sessionManager.authorizeSession(ctx.packageName, ctx).get()
+        return gson.fromJson(JSONObject(data).toString(), SFAKey::class.java)
+    }
 
-        val details: NodeDetails =
+    private fun getTorusNodeEndpoints(nodeDetails: NodeDetails): Array<String?> {
+        return if (this.network.toString().contains("sapphire")) {
+            nodeDetails.torusNodeSSSEndpoints
+        } else {
+            nodeDetails.torusNodeEndpoints
+        }
+    }
+
+    fun isSessionIdExists(): Boolean {
```

Comment:
what's the point of this?
if sessionId is expired? what will this help with?

Reply:
This is for checking whether sessionId exists or not.